plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchTaskHandler.groovy

```diff
@@ -277,21 +277,43 @@ class GoogleBatchTaskHandler extends TaskHandler implements FusionAwareTask {
 
         // use disk directive for an attached disk if type is specified
         if( disk?.type ) {
-            instancePolicy.addDisks(
-                AllocationPolicy.AttachedDisk.newBuilder()
-                    .setNewDisk(
-                        AllocationPolicy.Disk.newBuilder()
-                            .setType(disk.type)
-                            .setSizeGb(disk.request.toGiga())
-                    )
-                    .setDeviceName('scratch')
-            )
+            if ( disk?.image ) {
+                instancePolicy.addDisks(
+                    AllocationPolicy.AttachedDisk.newBuilder()
+                        .setNewDisk(
+                            AllocationPolicy.Disk.newBuilder()
+                                .setType(disk.type)
+                                .setImage(disk.image)
```

Comment:
If I understand correctly, when the image is specified the size is irrelevant?

Reply:
Yes, When image or snapshot are specified as a source, then the disk size is ignored.
I kept it in the example documentation as I cannot leave it empty or I would get an error "Not a valid disk value: null"

Reply:
And is there a reason why the type is also required when using an image?

Reply:
Disk type is not implied from the Image and would still be needed.
We can also add a default disk type if not specified by the user if you think this would be a better experience

Reply:
Just making sure I understand the requirements from google batch. Actually Nextflow uses the disk type to decide whether to apply the disk directive to the boot disk size or a new disk, so this should be fine as it is.